tests/Unit/DSCResources/DSC_PostgreSqlScript.tests.ps1

```diff
@@ -0,0 +1,196 @@
+                    $dscResult.DatabaseName | Should -Be $invalidParams.DatabaseName
+                    $dscResult.GetResult | Should -Be $null
+                }
+            }
+        }
+
+
+        Describe "$moduleResourceName\Set-TargetResource" -Tag 'Set'{
+            Context 'When Set-TargetResource runs successfully' {
+                BeforeAll {
+                    Mock -CommandName Invoke-Command -MockWith {}
+                }
+                Context 'When database does not exist' {
+                    BeforeEach {
+                        Mock -CommandName Invoke-Command -MockWith {}
+                        Mock Invoke-Command -Verifiable -ParameterFilter {$ScriptBlock -match '-lqt 2>&1'} -MockWith { return $psqlListResults }
+                    }
+                    It 'Should invoke psql and call CREATE DATABASE by default' {
+                        Mock -CommandName Invoke-Command -Verifiable -MockWith {return ""} -ParameterFilter {$ScriptBlock -match 'CREATE DATABASE'}
+
+                        Set-TargetResource @scriptParams
+                        Assert-MockCalled Invoke-Command -Exactly -Times 3 -Scope It
+                        Assert-VerifiableMock
+                    }
+
+                    It 'Should invoke psql and not create database when CreateDatabase parameter is false' {
+                        $NoCreateParams = $scriptParams.Clone()
+                        $NoCreateParams.CreateDatabase = $false
+
+                        Set-TargetResource @NoCreateParams
+                        Assert-VerifiableMock
+                        Assert-MockCalled Invoke-Command -Exactly -Times 2 -Scope It
+                        Assert-MockCalled Invoke-Command -Times 0 -Scope It -ParameterFilter {$ScriptBlock -match "CREATE DATABASE"}
+                    }
+                }
+
+                Context 'When database exists' {
+                    It 'Should not call CREATE DATABASE' {
+                        $psqlListResultsWithDatabase = @(" $($scriptParams.DatabaseName)      | postgres | UTF8")
+                        Mock Invoke-Command -Verifiable -ParameterFilter {$ScriptBlock -match '-lqt 2>&1'} -MockWith { return $psqlListResultsWithDatabase }
+
+                        Set-TargetResource @scriptParams
+                        Assert-VerifiableMock
+                        Assert-MockCalled Invoke-Command -Exactly -Times 2 -Scope It
+                        Assert-MockCalled Invoke-Command -Times 0 -Scope It -ParameterFilter {$ScriptBlock -match "CREATE DATABASE"}
+                    }
+                }
+            }
+
+            Context 'When Set-TargetResource fails' {
+                It 'Should throw when psql is not found' {
+                    $invalidParams = $scriptParams.Clone()
+                    $invalidParams.PsqlLocation = "Z:\does-not-exist.exe"
+
+                    {Set-TargetResource @invalidParams } | Should throw "is not recognized as the name of a cmdlet, function"
```

Comment:
I believe this should be -Throw -ExpectedMessage 'is not recognized as the name of a cmdlet, function'

Reply:
Good catch